lightningd/peer_control.c

```diff
@@ -49,6 +49,8 @@ static void destroy_peer(struct peer *peer)
 	list_del_from(&peer->ld->peers, &peer->list);
 	if (peer->fd >= 0)
 		close(peer->fd);
+	if (peer->gossip_client_fd >= 0)
+		close(peer->gossip_client_fd);
```

Comment:
As mentioned before this closes the socketpair and should also result in the peer-specific data to be freed on `gossipd`, nudge me if that isn't the case and I'll look into it.